npm/npm.go

```diff
@@ -271,53 +269,8 @@ func NewNetworkPolicyManager(clientset *kubernetes.Clientset, informerFactory in
 	// create NameSpace controller
 	npMgr.nameSpaceController = NewNameSpaceController(nsInformer, clientset, npMgr)
 
-	npInformer.Informer().AddEventHandler(
-		// Network policy event handlers
-		cache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj interface{}) {
-				networkPolicyObj, ok := obj.(*networkingv1.NetworkPolicy)
-				if !ok {
-					metrics.SendErrorLogAndMetric(util.NpmID, "ADD Network Policy: Received unexpected object type: %v", obj)
-					return
-				}
-				npMgr.Lock()
-				npMgr.AddNetworkPolicy(networkPolicyObj)
-				npMgr.Unlock()
-			},
-			UpdateFunc: func(old, new interface{}) {
-				oldNetworkPolicyObj, ok := old.(*networkingv1.NetworkPolicy)
-				if !ok {
-					metrics.SendErrorLogAndMetric(util.NpmID, "UPDATE Network Policy: Received unexpected old object type: %v", oldNetworkPolicyObj)
-					return
-				}
-				newNetworkPolicyObj, ok := new.(*networkingv1.NetworkPolicy)
-				if !ok {
-					metrics.SendErrorLogAndMetric(util.NpmID, "UPDATE Network Policy: Received unexpected new object type: %v", newNetworkPolicyObj)
-					return
-				}
-				npMgr.Lock()
-				npMgr.UpdateNetworkPolicy(oldNetworkPolicyObj, newNetworkPolicyObj)
-				npMgr.Unlock()
-			},
-			DeleteFunc: func(obj interface{}) {
-				networkPolicyObj, ok := obj.(*networkingv1.NetworkPolicy)
-				if !ok {
-					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-					if !ok {
-						metrics.SendErrorLogAndMetric(util.NpmID, "DELETE Network Policy: Received unexpected object type: %v", obj)
-						return
-					}
-					if networkPolicyObj, ok = tombstone.Obj.(*networkingv1.NetworkPolicy); !ok {
-						metrics.SendErrorLogAndMetric(util.NpmID, "DELETE Network Policy: Received unexpected object type: %v", obj)
-						return
-					}
-				}
-				npMgr.Lock()
-				npMgr.DeleteNetworkPolicy(networkPolicyObj)
-				npMgr.Unlock()
-			},
-		},
-	)
+	// create network policy controller
+	npMgr.netPolController = NewNetworkPolicyController(informerFactory.Networking().V1().NetworkPolicies(), clientset, npMgr)
```

Comment:
I would like to remove npInformer from NetworkPolicyManager struct in npm.go. Let me first check whether I can remove npInformer member. If not, I can use npInfomer.

Reply:
npInformer member in NetworkPolicyManager struct  is also used in Start method while it may be not necessary to keep the member. I changed it to "npInformer".